src/button.ts

```diff
@@ -0,0 +1,23 @@
+import { HTML } from '@bigtest/interactor';
+
+
+export default HTML.extend<HTMLButtonElement | HTMLLinkElement>('MUI Button')
+  .selector('.MuiButton-root')
+  .filters({
+    iconLabel: (el) => el.querySelector(".MuiButton-label")?.firstElementChild?.classList.value,
+    customRippleClass: (el) => {
+      const classes: string[] = []
+      const spans = el.querySelectorAll('span');
+      spans.forEach((span) => {
+        classes.push(span.classList.value)
+      });
+
+      return classes.join(" ")
+    },
+    href: (el) => el.getAttribute('href'),
+    disabled: (el) => el.disabled,
+    rippleEnabled: (el) => !!el.querySelector(".MuiTouchRipple-root"),
+    touchRipple: (el) => !!(el).querySelector(".MuiTouchRipple-ripple"),
+    rippleVisible: (el) => !!(el).querySelector(".MuiTouchRipple-rippleVisible")
```

Comment:
What do these filters mean?

Reply:
They mosttly have to do with the ripple animation on the button if I remember correctly. But how the classes are put on the elements varies, if you pass an object it does one thing, if you pass a prop another thing etc. Do you think we need to have these filters or no?

Reply:
I think it's one of those things where consumers of material-ui components won't need them in their tests, but it's the kind of thing that _authors_ of material-ui will want to test, so I would just just use a `.has({ className: including('rippleVisible') })` assertion instead of creating a filter.